pkg/lib/feature/stdattrs/service_noevent.go

```diff
@@ -193,63 +194,102 @@ func (s *ServiceNoEvent) UpdateStandardAttributes(role accesscontrol.Role, userI
 	return nil
 }
 
-// DeriveStandardAttributes populates email_verified and phone_number_verified,
-// if email or phone_number are found in attrs.
-func (s *ServiceNoEvent) DeriveStandardAttributes(role accesscontrol.Role, userID string, updatedAt time.Time, attrs map[string]interface{}) (map[string]interface{}, error) {
-	out := make(map[string]interface{})
+// Batch implementation of DeriveStandardAttributes
+func (s *ServiceNoEvent) DeriveStandardAttributesForUsers(
+	role accesscontrol.Role,
+	userIDs []string,
+	updatedAts []time.Time,
+	attrsList []map[string]interface{},
+) (map[string]map[string]interface{}, error) {
 
-	for key, value := range attrs {
-		value, err := s.Transformer.StorageFormToRepresentationForm(key, value)
-		if err != nil {
-			return nil, err
-		}
+	if len(userIDs) != len(updatedAts) || len(userIDs) != len(attrsList) {
+		panic("stdattrs: expeceted same length of arguments")
+	}
 
-		// Copy
-		out[key] = value
+	allClaims, err := s.ClaimStore.ListByUserIDsAndClaimNames(
```

Comment:
This slice contains claims of different users. If there are 100 hundreds, and each user has an email address and a phone number, this slice is of length 200. And the the following loop has at most 100 * 200 iterations.

Maybe we can group `allClaims` into `claimsByUserID map[string][]*Claim` first? Then we need 200 iterations to group and 100 iterations to build the results.